app/scripts/controllers/monitoring.js

```diff
@@ -0,0 +1,311 @@
+'use strict';
+/* jshint unused: false */
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:MonitoringController
+ * @description
+ * # MonitoringController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('MonitoringController', function ($routeParams,
+                                           $scope,
+                                           $filter,
+                                           DataService,
+                                           ProjectsService,
+                                           MetricsService,
+                                           BuildsService,
+                                           PodsService,
+                                           Logger,
+                                           ImageStreamResolver,
+                                           $rootScope) {
+    $scope.projectName = $routeParams.project;
+    $scope.alerts = $scope.alerts || {};
+    $scope.renderOptions = $scope.renderOptions || {};
+    $scope.renderOptions.showEventsSidebar = true;
+    $scope.renderOptions.collapseEventsSidebar = localStorage.getItem('monitoring.eventsidebar.collapsed') === 'true';
+
+
+    var watches = [];
+
+    $scope.kindSelector = {
+      selected: {
+        kind: "All"
+      }
+    };
+    $scope.kinds = [
+      {
+        kind: "Pods"
+      },
+      {
+        kind: "Builds"
+      },
+      {
+        label: "Deployments",
+        kind: "ReplicationControllers"
+      },
+      {
+        kind: "All"
+      }
+    ];
+
+    $scope.logOptions = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.logCanRun = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.logEmpty = {
+      pods: {},
+      deployments: {},
+      builds: {}
+    };
+
+    $scope.filters = {
+      hideOlderResources: true,
+      text: ''
+    };
+
+    var ageFilteredBuilds, ageFilteredDeployments, ageFilteredPods;
+
+    // Check if the metrics service is available so we know when to show the tab.
+    MetricsService.isAvailable().then(function(available) {
+      $scope.metricsAvailable = available;
+    });
+
+    var orderByDate = $filter('orderObjectsByDate');
+
+    var filterExpressions = [];
+    var updateKeywords = function() {
+      if (!$scope.filters.text) {
+        filterExpressions = [];
+        return;
+      }
+
+      var keywords = _.uniq($scope.filters.text.split(/\s+/));
+      // Sort the longest keyword first.
+      keywords.sort(function(a, b){
+        return b.length - a.length;
+      });
+
+      // Convert the keyword to a case-insensitive regular expression for the filter.
+      filterExpressions = _.map(keywords, function(keyword) {
+        return new RegExp(_.escapeRegExp(keyword), "i");
+      });
```

Comment:
Might want to pull some of this logic into a service so it can be reused anywhere we add a text filter (monitoring, events, project list).